# Trabajo Final de "PROGRAMACIÓN 1" - TUIA - 2C - 2023
# Integrantes del Grupo:
*   XXXX, XXXXX
*   XXXX, XXXXX
*   XXXX, XXXXX
*   XXXX, XXXXX








In [3]:
#! pip3 install matplotlib.pyplot
!# pip3 install seaborn

In [4]:
#import matplotlib.pyplot as plt
#import seaborn as sns

# Presentación del tema: **Servicio Meteorológico**

El **Servicio Meteorológico Nacional** publica diferentes mediciones que se realizan de temperaturas, viento, humedad, precipitaciones, etc. en diferentes lugares del país en la [web](https://datos.gob.ar/dataset?groups=envi&organization=defensa&_tags_limit=0).


En el presente trabajo vamos a utilizar los datos publicados sobre [mediciones](https://drive.google.com/file/d/1HcjvyLIbMNk4GceqhLDcUkg--Rv0vpOU/view?usp=sharing) efectuadas en diferentes
puntos del país en el último año.

# Estructura de datos

Para la resolución del presente trabajo debe usar alguna de las estructuras de datos vistas en la materia:

*   Listas
*   Diccionarios
*   Tuplas
*   Clases y Objetos

o una combinación de ellas.







# Carga de datos en las estructuras

En grupo, analicen el contenido del archivo *servicio_meteorológico.csv* que provee ciertas mediciones efectuadas en diferentes estaciones meteorológicas del país.

*Para la implementación de este trabajo final, recuerden que es muy importante comentar el código, usar nombres de variables significativos y definir funciones para trabajar de manera modular.*



In [5]:
#Las siguientes líneas de código son necesarias para poder trabajar con archivos .csv y realizar gráficos
import csv
#import matplotlib.pyplot as plt
#import seaborn as sns
#sns.set()

NOTA: en clase se explicará cómo trabajar con archivos .csv y cargar datos en alguna de las estructuras de datos vistas durante la materia. Luego, en grupo, deberán analizar cuáles son las estructuras de datos más convenientes según las preguntas que deban responder más adelante. Completaremos el enunciado en las próximas semanas. Mientras tanto, analicen los datos del archivo compartido y practiquen la lectura de datos desde archivos implementando una o varias funciones carga (para las distintas estructuras de datos que propongan).

In [145]:
def openFile(*args) -> list:
    """
    Recibe un archivo de tipo csv y devuelve dos listas, una los datos de cada estación de observación meteorológica y otra el encabezado.
    """
    with open(*args, "r", encoding="macintosh") as contenido:
        lector = csv.reader(contenido)
        lista_retorno = []
        encabezado = next(lector)
        for fila in lector:
            for elemento in fila:
                lista_retorno.append(elemento)

    return lista_retorno, encabezado

In [146]:
#definan una o varias funciones para imprimir la información cargada
#reemplace el nombre imprimir_xxx por un nombre adecuado
def imprimir_xxx() -> None: #completar el tipo de parámetro
	#toma como parámetro una estructura de datos y muestra
	#en pantalla su contenido de manera elegante

  #completar
    pass
#usen esta(s) función(es) para chequear que la función carga hace lo que se espera

# Funciones


1.   Crear UNA función que sirva para calcular el promedio, el máximo y el mínimo anual de cada medida, asumiendo que todos los datos están completos y son correctos.
2.   Modifique la función anterior, sabiendo que no se disponen de todas las mediciones. Aquellas mediciones faltantes se indentifican con 'S/D' y no deben ser tenidas en cuenta para los cálculos.


In [152]:
def medidasTendencia(lista_retorno: list, encabezado: list) -> list:
    cant_columnas = 14
    columna = []
    suma = 0
    for elemento in lista_retorno:
        columna.append(elemento)
        if len(columna) == cant_columnas:
            print(columna)
            lista = []            
            for elemento_interno in range(len(columna)):                
                try:
                    columna[elemento_interno] = float(columna[elemento_interno])                    
                    suma += columna[elemento_interno]
                    promedio = suma / 12
                    lista.append(columna[elemento_interno])                        
                    maximo = max(lista)
                    minimo = min(lista)
                except:
                    print("Valor salteado")
                    continue
            print("El promedio es de: ", promedio, "El máximo es de: ", maximo, "El mínimo es de: ", minimo, ".")
            columna = []
        suma = 0
           
    return

## Objetos


1. Cree una clase llamada `EstacionMeteorologica` que represente una estación meteorológica. La clase debe tener como atributos de instancia los datos correspondientes a:
* nombre de la estacion (nombre_estacion)
* temperaturas promedios para cada mes del año (temps_prom_mens)
* temperatura maxima anual (temp_maxima)
* temperatura minima anual (temp_minima)
* humedad anual promedio (hum_promedio)
* humedad maxima anual (hum_maxima)
* humedad minima anual (hum_minima)
* velocidad promedio anual (vel_promedio)
* velocidad maxima anual (vel_maxima)
* velocidad minima anual (vel_minima)


2.   Implemente el método **\_\_init\_\_** y **\_\_str\_\_**.
# Objetos


1. Cree una clase llamada **EstacionMeteorologica** que represente una estación meteorológica. La clase debe tener como atributos de instancia los datos correspondientes a:
  * nombre de la estacion (**nombre_estacion**)
  * temperaturas promedios para cada mes del año (**temps_prom_mens**)
  * temperatura maxima anual (**temp_maxima**)
  * temperatura minima anual (**temp_minima**)
  * humedad anual promedio (**hum_promedio**)
  * humedad maxima anual (**hum_maxima**)
  * humedad minima anual (**hum_minima**)
  * velocidad promedio anual (**vel_promedio**)
  * velocidad maxima anual (**vel_maxima**)
  * velocidad minima anual (**vel_minima**)

2. Implemente el método **\_\_init\_\_** y **\_\_str\_\_**.


3. Implemente los siguientes métodos en la clase **EstacionMeteorologica**:

  3.1  Un método **promedio_temp_anual** que calcule el promedio de temperatura anual para la estación

  3.1. Un método **comparar_con** que reciba otra estación como parámetro y retorne una comparación mes a mes de las temperaturas promedio entre las dos estaciones.

  3.2. Un método **graficar_temperaturas** que grafique la evolución de las temperaturas en la estación a lo largo del año


In [153]:
class EstacionMeteorologica:
    """
    Esta clase representa una estación meteorológica,
    contiene los datos relevantes al análisis.
    """
    def __init__(self, nombre_estacion, temps_prom_mens, temp_maxima, temp_minima,
                 hum_promedio, hum_maxima, hum_minima, vel_promedio, vel_maxima, vel_minima):
        self.nombre_estacion = nombre_estacion
        self.temps_prom_mens = temps_prom_mens
        self.temp_maxima = temp_maxima
        self.temp_minima = temp_minima
        self.hum_promedio = hum_promedio
        self.hum_maxima = hum_maxima
        self.hum_minima = hum_minima
        self.vel_promedio = vel_promedio
        self.vel_maxima = vel_maxima
        self.vel_minima = vel_minima

    def __str__(self):
        return f"El presente objeto representa a la estación meteorológica {self.nombre_estacion}. Cada estación provee información sobre {self.temps_prom_mens}, {self.temps_prom_mens}, {self.temp_maxima}, {self.temp_minima}, {self.hum_promedio}, {self.hum_maxima}, {self.hum_minima}, {self.vel_promedio}, {self.vel_maxima}, {self.vel_minima}."

In [154]:
def instanciaEstacion(lista_retorno: list) -> list:
    """
    La función instanciaViaje crea instancias a partir del output lista_retorno creado por la función openFile. Primero
    inicializa una variable fija de contador de cantidad de atributos totales de la clase Viaje. Luego se crea una lista
    de listas, que en su interior contiene las instancias individuales de cada Viaje. Se itera un bucle que toma los
    elementos de la lista y al igualar el número de atributos de la instancia, los atribuye a una instancia individual
    del objeto. Este objeto se añade a la lista de listas y se reinicia la lista de objetos individuales inicializándose
    vacía. Por último, retorna una lista que contiene todas las instancias de Viaje.
    """
    nro_de_atributos_instancia = 14
    instancia = []
    instancia_de_estaciones = []

    for elemento in lista_retorno:
        instancia.append(elemento)
        if len(instancia) == nro_de_atributos_instancia:
            estacion = EstacionMeteorológica(instancia[0], instancia[1], instancia[2], int(instancia[3]), int(instancia[4]), instancia[5])
            instancia_de_viajes.append(estacion)
            instancia = []
    print("Se crearon ", len(instancia_de_estaciones), "instancias de estaciones")
    return instancia_de_viajes

In [155]:
estaciones, encabezado = openFile("servicio_meteorologico.csv")

#print(estaciones)

medidasTendencia(estaciones, encabezado)

['LA QUIACA OBS.', 'Temperatura (grados C)', '12.8', '12.5', '12.4', '10.9', '6.9', '4.4', '4.1', '6.7', '9.3', '11.7', '12.8', '13.2']
Valor salteado
Valor salteado
El promedio es de:  9.808333333333334 El máximo es de:  13.2 El mínimo es de:  4.1 .
['LA QUIACA OBS.', 'Temperatura máxima (grados C)', '20.1', '19.9', '19.9', '19.7', '17.3', '15.6', '15.4', '17.4', '19.2', '21.1', '21.7', '21.3']
Valor salteado
Valor salteado
El promedio es de:  19.049999999999997 El máximo es de:  21.7 El mínimo es de:  15.4 .
['LA QUIACA OBS.', 'Temperatura mínima (grados C)', '7.4', '7.0', '6.3', '2.8', '-3.0', '-6.2', '-6.8', '-4.2', '-1.1', '2.8', '5.2', '6.9']
Valor salteado
Valor salteado
El promedio es de:  1.425 El máximo es de:  7.4 El mínimo es de:  -6.8 .
['LA QUIACA OBS.', 'Humedad relativa (%)', '65.3', '64.1', '62.4', '48.2', '34.6', '31.1', '30.1', '31.7', '34.4', '45.3', '52.3', '59.2']
Valor salteado
Valor salteado
El promedio es de:  46.55833333333334 El máximo es de:  65.3 El mínimo 

El promedio es de:  3.8416666666666672 El máximo es de:  4.4 El mínimo es de:  3.6 .
['FORMOSA AERO', 'Precipitación (mm)', '165.7', '128.7', '144.8', '175.0', '99.3', '68.9', '47.5', '44.2', '90.8', '142.0', '166.5', '163.4']
Valor salteado
Valor salteado
El promedio es de:  119.73333333333335 El máximo es de:  175.0 El mínimo es de:  44.2 .
['FORMOSA AERO', 'Frecuencia de días con Precipitación superior a 0.1 mm', '9.1', '8.9', '8.5', '9.2', '7.7', '7.2', '5.5', '5.8', '7.6', '9.6', '9.1', '9.1']
Valor salteado
Valor salteado
El promedio es de:  8.108333333333333 El máximo es de:  9.6 El mínimo es de:  5.5 .
['CORRIENTES AERO', 'Temperatura (grados C)', '27.0', '26.0', '24.7', '21.4', '18.0', '15.9', '15.2', '17.0', '18.5', '21.8', '23.7', '25.8']
Valor salteado
Valor salteado
El promedio es de:  21.25 El máximo es de:  27.0 El mínimo es de:  15.2 .
['CORRIENTES AERO', 'Temperatura máxima (grados C)', '33.2', '31.9', '30.6', '26.8', '23.4', '21.0', '21.1', '23.4', '24.9', '27.9', '29

['VILLA MARIA DEL RIO SECO', 'Humedad relativa (%)', '74.7', '79.7', '81.8', '82.1', '79.0', '76.7', '71.8', '66.3', '64.7', '67.5', '70.4', '72.5']
Valor salteado
Valor salteado
El promedio es de:  73.93333333333332 El máximo es de:  82.1 El mínimo es de:  64.7 .
['VILLA MARIA DEL RIO SECO', 'Velocidad del Viento (km/h)', '6.2', '5.7', '5.2', '5.3', '5.2', '5.3', '5.7', '6.8', '7.6', '8.8', '8.6', '7.1']
Valor salteado
Valor salteado
El promedio es de:  6.458333333333332 El máximo es de:  8.8 El mínimo es de:  5.2 .
['VILLA MARIA DEL RIO SECO', 'Nubosidad total (octavos)', '3.9', '4.1', '4.1', '4.4', '4.4', '4.2', '3.3', '3.2', '3.3', '3.6', '3.8', '4.0']
Valor salteado
Valor salteado
El promedio es de:  3.858333333333333 El máximo es de:  4.4 El mínimo es de:  3.2 .
['VILLA MARIA DEL RIO SECO', 'Precipitación (mm)', '130.6', '119.7', '134.5', '70.4', '21.3', '8.4', '12.5', '6.7', '27.1', '62.6', '89.0', '130.2']
Valor salteado
Valor salteado
El promedio es de:  67.75 El máximo es de:

El promedio es de:  61.76666666666667 El máximo es de:  72.2 El mínimo es de:  50.5 .
['VILLA DOLORES AERO', 'Velocidad del Viento (km/h)', '7.1', '5.0', '4.6', '5.0', '4.6', '4.3', '5.6', '6.5', '7.1', '8.7', '8.6', '7.3']
Valor salteado
Valor salteado
El promedio es de:  6.199999999999999 El máximo es de:  8.7 El mínimo es de:  4.3 .
['VILLA DOLORES AERO', 'Nubosidad total (octavos)', '3.4', '3.5', '3.6', '3.5', '3.8', '3.6', '3.1', '3.0', '3.1', '3.2', '3.3', '3.5']
Valor salteado
Valor salteado
El promedio es de:  3.3833333333333333 El máximo es de:  3.8 El mínimo es de:  3.0 .
['VILLA DOLORES AERO', 'Precipitación (mm)', '121.1', '102.2', '94.4', '42.4', '17.6', '4.7', '13.2', '6.1', '22.9', '35.1', '75.1', '113.3']
Valor salteado
Valor salteado
El promedio es de:  54.00833333333333 El máximo es de:  121.1 El mínimo es de:  4.7 .
['VILLA DOLORES AERO', 'Frecuencia de días con Precipitación superior a 0.1 mm', '10.5', '9.1', '8.6', '5.5', '3.7', '1.8', '2.5', '1.6', '3.7', '4.8', '

El promedio es de:  74.09166666666665 El máximo es de:  81.9 El mínimo es de:  65.8 .
['MONTE CASEROS AERO', 'Velocidad del Viento (km/h)', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D']
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
El promedio es de:  74.09166666666665 El máximo es de:  81.9 El mínimo es de:  65.8 .
['MONTE CASEROS AERO', 'Nubosidad total (octavos)', '3.3', '3.5', '3.2', '3.6', '3.6', '4.2', '3.7', '3.7', '3.8', '3.6', '3.3', '3.3']
Valor salteado
Valor salteado
El promedio es de:  3.566666666666666 El máximo es de:  4.2 El mínimo es de:  3.2 .
['MONTE CASEROS AERO', 'Precipitación (mm)', '154.1', '144.7', '154.3', '189.7', '110.9', '88.4', '64.1', '57.5', '107.3', '146.5', '154.0', '134.1']
Valor salteado
Valor salteado
El promedio es de:  125.46666666666665 El máximo e

['VILLA REYNOLDS AERO', 'Velocidad del Viento (km/h)', '12.4', '11.3', '10.2', '8.9', '8.5', '8.4', '8.9', '10.5', '12.3', '15.1', '15.8', '14.3']
Valor salteado
Valor salteado
El promedio es de:  11.383333333333333 El máximo es de:  15.8 El mínimo es de:  8.4 .
['VILLA REYNOLDS AERO', 'Nubosidad total (octavos)', '3.8', '3.6', '3.7', '3.6', '4.1', '4.2', '3.8', '3.7', '3.8', '4.0', '3.9', '3.9']
Valor salteado
Valor salteado
El promedio es de:  3.8416666666666663 El máximo es de:  4.2 El mínimo es de:  3.6 .
['VILLA REYNOLDS AERO', 'Precipitación (mm)', '103.7', '85.7', '94.6', '55.0', '28.5', '9.5', '18.8', '17.1', '35.2', '58.8', '85.9', '113.1']
Valor salteado
Valor salteado
El promedio es de:  58.82500000000001 El máximo es de:  113.1 El mínimo es de:  9.5 .
['VILLA REYNOLDS AERO', 'Frecuencia de días con Precipitación superior a 0.1 mm', '9.8', '8.1', '9.0', '6.0', '4.4', '2.7', '3.7', '3.4', '5.4', '8.3', '9.7', '10.9']
Valor salteado
Valor salteado
El promedio es de:  6.7833333

El promedio es de:  58.68333333333334 El máximo es de:  68.0 El mínimo es de:  50.2 .
['SAN RAFAEL AERO', 'Nubosidad total (octavos)', '2.7', '2.7', '2.9', '2.9', '3.4', '3.4', '3.2', '3.0', '3.1', '3.0', '2.8', '2.6']
Valor salteado
Valor salteado
El promedio es de:  2.975 El máximo es de:  3.4 El mínimo es de:  2.6 .
['SAN RAFAEL AERO', 'Precipitación (mm)', '61.0', '42.8', '37.2', '26.1', '15.6', '9.1', '13.2', '16.0', '27.4', '30.5', '38.6', '45.3']
Valor salteado
Valor salteado
El promedio es de:  30.233333333333334 El máximo es de:  61.0 El mínimo es de:  9.1 .
['SAN RAFAEL AERO', 'Frecuencia de días con Precipitación superior a 0.1 mm', '7.5', '6.1', '5.3', '4.1', '3.7', '2.9', '2.8', '3.3', '4.7', '4.7', '5.7', '6.7']
Valor salteado
Valor salteado
El promedio es de:  4.791666666666667 El máximo es de:  7.5 El mínimo es de:  2.8 .
['GENERAL PICO AERO', 'Temperatura (grados C)', '23.8', '22.5', '20.2', '16.0', '12.0', '8.8', '8.1', '10.4', '13.1', '16.7', '20.0', '22.8']
Valor sa

El promedio es de:  89.30000000000001 El máximo es de:  128.6 El mínimo es de:  51.4 .
['SAN MIGUEL', 'Frecuencia de días con Precipitación superior a 0.1 mm', '8.4', '8.2', '8.6', '9.9', '7.1', '7.5', '7.3', '6.7', 'S/D', '9.9', '9.9', '9.2']
Valor salteado
Valor salteado
Valor salteado
El promedio es de:  7.725000000000001 El máximo es de:  9.9 El mínimo es de:  6.7 .
['EL PALOMAR AERO', 'Temperatura (grados C)', '24.1', '22.9', '20.9', '16.7', '13.1', '10.2', '9.6', '11.6', '13.6', '16.8', '19.7', '22.4']
Valor salteado
Valor salteado
El promedio es de:  16.8 El máximo es de:  24.1 El mínimo es de:  9.6 .
['EL PALOMAR AERO', 'Temperatura máxima (grados C)', '29.9', '28.5', '26.5', '22.5', '18.9', '15.7', '15.1', '17.4', '19.2', '22.3', '25.2', '28.3']
Valor salteado
Valor salteado
El promedio es de:  22.458333333333332 El máximo es de:  29.9 El mínimo es de:  15.1 .
['EL PALOMAR AERO', 'Temperatura mínima (grados C)', '18.1', '17.3', '15.5', '11.4', '7.9', '5.5', '4.8', '6.2', '8.0'

El promedio es de:  15.466666666666669 El máximo es de:  23.6 El mínimo es de:  7.2 .
['SANTA ROSA AERO', 'Temperatura máxima (grados C)', '31.0', '29.8', '26.9', '22.4', '18.0', '14.7', '14.2', '17.2', '19.7', '23.1', '26.7', '29.8']
Valor salteado
Valor salteado
El promedio es de:  22.79166666666666 El máximo es de:  31.0 El mínimo es de:  14.2 .
['SANTA ROSA AERO', 'Temperatura mínima (grados C)', '16.4', '15.1', '13.4', '9.2', '5.7', '3.0', '2.0', '3.4', '5.7', '9.2', '12.2', '15.1']
Valor salteado
Valor salteado
El promedio es de:  9.200000000000001 El máximo es de:  16.4 El mínimo es de:  2.0 .
['SANTA ROSA AERO', 'Humedad relativa (%)', '58.6', '62.6', '69.6', '72.1', '75.4', '75.9', '72.7', '64.8', '63.0', '62.1', '57.6', '55.4']
Valor salteado
Valor salteado
El promedio es de:  65.81666666666666 El máximo es de:  75.9 El mínimo es de:  55.4 .
['SANTA ROSA AERO', 'Velocidad del Viento (km/h)', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D']
V

El promedio es de:  77.17499999999998 El máximo es de:  95.7 El mínimo es de:  55.7 .
['MAR DEL PLATA AERO', 'Frecuencia de días con Precipitación superior a 0.1 mm', '8.3', '7.9', '9.0', '9.1', '8.6', '8.5', '8.8', '7.9', '7.9', '9.8', '9.6', '8.8']
Valor salteado
Valor salteado
El promedio es de:  8.683333333333334 El máximo es de:  9.8 El mínimo es de:  7.9 .
['NEUQUEN AERO', 'Temperatura (grados C)', '23.7', '22.3', '18.9', '13.4', '9.2', '6.6', '5.8', '8.1', '11.4', '15.6', '19.4', '22.4']
Valor salteado
Valor salteado
El promedio es de:  14.733333333333334 El máximo es de:  23.7 El mínimo es de:  5.8 .
['NEUQUEN AERO', 'Temperatura máxima (grados C)', '31.9', '30.6', '27.0', '21.6', '16.5', '12.9', '12.9', '15.9', '18.9', '23.2', '27.1', '30.2']
Valor salteado
Valor salteado
El promedio es de:  22.391666666666666 El máximo es de:  31.9 El mínimo es de:  12.9 .
['NEUQUEN AERO', 'Temperatura mínima (grados C)', '15.5', '14.2', '11.6', '6.8', '3.5', '1.5', '0.2', '1.6', '4.4', '8.2'

El promedio es de:  3.9000000000000004 El máximo es de:  4.7 El mínimo es de:  3.0 .
['ESQUEL AERO', 'Precipitación (mm)', '18.8', '18.4', '24.5', '43.1', '65.0', '79.4', '60.2', '47.0', '35.7', '28.3', '20.8', '20.8']
Valor salteado
Valor salteado
El promedio es de:  38.50000000000001 El máximo es de:  79.4 El mínimo es de:  18.4 .
['ESQUEL AERO', 'Frecuencia de días con Precipitación superior a 0.1 mm', '3.8', '3.7', '5.6', '7.0', '9.7', '11.0', '9.4', '9.6', '7.3', '6.1', '4.4', '4.1']
Valor salteado
Valor salteado
El promedio es de:  6.808333333333333 El máximo es de:  11.0 El mínimo es de:  3.7 .
['TRELEW AERO', 'Temperatura (grados C)', '21.5', '20.3', '17.6', '13.3', '9.2', '6.2', '5.8', '7.7', '10.4', '14.1', '17.4', '20.0']
Valor salteado
Valor salteado
El promedio es de:  13.625 El máximo es de:  21.5 El mínimo es de:  5.8 .
['TRELEW AERO', 'Temperatura máxima (grados C)', '29.2', '28.1', '25.0', '20.7', '15.7', '12.2', '12.3', '14.8', '17.5', '21.2', '24.9', '27.5']
Valor sa

Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
El promedio es de:  5.625 El máximo es de:  6.3 El mínimo es de:  5.0 .
['BASE ESPERANZA', 'Frecuencia de días con Precipitación superior a 0.1 mm', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D']
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
El promedio es de:  5.625 El máximo es de:  6.3 El mínimo es de:  5.0 .
['BASE ORCADAS', 'Temperatura (grados C)', '1.4', '1.4', '0.4', '-1.8', '-4.6', '-7.9', '-9.3', '-7.8', '-5.4', '-2.8', '-0.7', '0.6']
Valor salteado
Valor salteado
El promedio es de:  -3.0416666666666665 El máximo es de:  1.4 El mínimo es de:  -9.3 .
['BASE ORCADAS', 'Tempe

Valor salteado
El promedio es de:  -1.3500000000000005 El máximo es de:  4.5 El mínimo es de:  -7.3 .
['BASE SAN MARTIN', 'Temperatura mínima (grados C)', '-0.3', '-1.3', '-3.6', '-5.8', '-8.4', '-13.5', '-16.3', '-16.8', '-13.9', '-10.2', '-5.5', '-1.8']
Valor salteado
Valor salteado
El promedio es de:  -8.116666666666667 El máximo es de:  -0.3 El mínimo es de:  -16.8 .
['BASE SAN MARTIN', 'Humedad relativa (%)', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D', 'S/D']
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
Valor salteado
El promedio es de:  -8.116666666666667 El máximo es de:  -0.3 El mínimo es de:  -16.8 .
['BASE SAN MARTIN', 'Velocidad del Viento (km/h)', '14.6', '17.7', '22.3', '20.4', '19.3', '18.0', '17.3', '18.9', '17.5', '20.3', '19.2', '18.3']
Valor salteado
Valor salteado
El promedio es de:  18.

In [120]:
lista = ['LA QUIACA OBS.', 'Temperatura m‡xima (grados C)', '20.1', '19.9', '19.9', '19.7', '17.3', '15.6', '15.4', '17.4', '19.2', '21.1', '21.7', '21.3']
print(max(lista))

Temperatura m‡xima (grados C)


4. Responda a las siguientes preguntas teniendo en cuenta los datos leídos del archivo (pueden agregar más métodos a la clase **EstacionMeteorologica** si lo necesitan para responder a las preguntas):

  4.1. ¿En qué mes y estación se registró la velocidad del viento más alta?

  4.2. ¿En qué mes y estación se registró la humedad relativa más baja?

  4.3 Realice un gráfico para comparar las temperaturas promedio anuales de todas las estaciones

5. En grupo, piensen qué pregunta pueden realizarle a los datos disponibles en el archivo y respóndanla. Debe ser una pregunta que pueda ser graficada. Pueden agregar más atributos a la clase **EstacionMeteorologica**, si así lo necesitan.